### DISCUSSION TITLE
Cannot read property '_x_refs' of undefined after upgrade to v3

### DISCUSSION THREAD

**Discussion by Laurentvw** [alpinejs/alpine#1568] [7. Bugs (archived)] [answered]
I am getting the following error after upgrading to v3(.0.8):

```
Alpine Expression Error: Cannot read property '_x_refs' of undefined

Expression: "true"

 <template x-if=​"true">​…​</template>​
```

Error line is in x-ref.js : 
```
let root = closestRoot(el);
    if (!root._x_refs) <- here
```

When I remove the x-ref which is present in the child element of the template element, the error disappears.
```
<template x-if="true">
    <ul x-ref="listbox">...</ul>
</template>
```

I had no errors when in v2.

Edit: added JSFiddle: https://jsfiddle.net/3bk7510s/2/

Could this be a bug? Any idea why it could be acting this way? Thanks.

**Comment by calebporzio** [comment, marked as answer]
Definitely a bug. Just merged the fix: https://github.com/alpinejs/alpine/pull/1574

Will be available in the next release. Thanks!

**Reply by ryangjchandler** [reply]
@Laurentvw Could you mark this discussion as answered please. :)

**Reply by Laurentvw** [reply]
Done! Thanks for fixing. Have a great day.
